### PR TITLE
Add Tornjak Tests

### PR DESCRIPTION
**Pull request by mrsabath**

### CODE REVIEW

File: .github/tests/tornjak/install.sh [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+SCRIPT=$(readlink -f "$0")
+SCRIPTPATH=$(dirname "$SCRIPT")
+
+helm install \
+  --namespace spire-server \
+  --values "${SCRIPTPATH}/../../../examples/production/values.yaml" \
+  --values "${SCRIPTPATH}/../../../examples/tornjak/values.yaml" \
+  spire charts/spire --wait
+  helm test spire -n spire-server
```
Comment (kfox1111) [inline .github/tests/tornjak/install.sh:13]:
```suggestion
helm test spire -n spire-server
```

File: examples/tornjak/values.yaml [hunk 2]
```diff
@@ -0,0 +1,47 @@
+spire-server:
+  tornjak:
+    enabled: true
+    image:
+      registry: ghcr.io
+      repository: spiffe/tornjak-be
+      pullPolicy: IfNotPresent
+      version: "v1.0.2"
```
Comment (kfox1111) [inline examples/tornjak/values.yaml:8]: Probably should leave out most of the default stuff? The chart readme should cover it?

Maybe leave in anything the user is very likely to need to override such as the apiServerURL?

File: examples/tornjak/values.yaml [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    fsGroup: 1000
+    fsGroupChangePolicy: OnRootMismatch
+  securityContext:
+    allowPrivilegeEscalation: false
+    runAsNonRoot: true
+    readOnlyRootFilesystem: false
+    capabilities:
+      drop: [ALL]
+    seccompProfile:
+      type: RuntimeDefault
```
Comment (kfox1111) [inline examples/tornjak/values.yaml:43]: Thinking this stuff should be put in the production examples values file. Then this values file becomes more about turning on/cofiguraing tornjak, while the other one is about productionizing all the settings for the possible features.

File: examples/tornjak/values.yaml (hunk 2 above)
Comment (kfox1111) [reply examples/tornjak/values.yaml:8]: oh. and apiServerURL here seems dead. Its the frontend that cares I think?

File: examples/tornjak/README.md [hunk 4]
```diff
@@ -0,0 +1,16 @@
+# Recommended setup to deploy Tornjak
+
+Too install Spire with the least privileges possible we deploy spire across 2 namespaces.
```
Comment (kfox1111) [inline examples/tornjak/README.md:3]:
```suggestion
To install Spire with the least privileges possible we deploy spire across 2 namespaces.
```

**Review by kfox1111 — COMMENTED** [review]
Looking pretty good. Some changes inline.

File: examples/tornjak/values.yaml (hunk 3 above)
Comment (mrsabath) [reply examples/tornjak/values.yaml:19]: So about that, when I move the `podSecurityContext` and `securityContext` values to _examples/production/values.yaml_ they got rendered as: 
```yaml
     securityContext:
         {}
```         
 All the other values work fine. I need to figure out why. Perhaps it has something to do how we format the values in the template?:
 ```yaml
       securityContext:
        {{- toYaml .Values.podSecurityContext | nindent 8 }}   
 ```

Comment (mrsabath) [reply examples/tornjak/values.yaml]: I tried again and it seems to work the second time around.

File: charts/spire/charts/tornjak-frontend/Chart.yaml [hunk 5]
```diff
@@ -4,6 +4,10 @@ description: A Helm chart to deploy Tornjak frontend
 type: application
 version: 0.1.0
 appVersion: "v1.0.2"
+home: https://github.com/spiffe/helm-charts/tree/main/charts/spire
+sources:
+  - https://github.com/spiffe/tornjak
+icon: https://github.com/spiffe/tornjak/blob/main/logos/logo%2Btornjak.2132x1291.png
```
Comment (kfox1111) [inline charts/spire/charts/tornjak-frontend/Chart.yaml:10]: I think the other url is to the web page with the image embedded. 
```suggestion
icon: https://raw.githubusercontent.com/spiffe/tornjak/main/logos/logo%2Btornjak.2132x1291.png
```

File: .github/tests/tornjak/post-install.sh [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+| spire-server                         | "$("${k_rollout_status[@]}" spire-server statefulset spire-server)" |
+| tornjak-frontend                     | "$("${k_wait[@]}" spire-server deployments.apps spire-tornjak-frontend)" |
+| tornjak-frontend                     | "$("${k_rollout_status[@]}" spire-server deployments.apps spire-tornjak-frontend)" |
+EOF
+
+kubectl -n spire-server get service spire-tornjak-frontend
+
+if [ $1 -ne 0 ]; then
+  get_namespace_details spire-server
+fi
```
Comment (kfox1111) [inline .github/tests/tornjak/post-install.sh:50]: Probably still want all the stuff in both namespaces, in case something fails?

**Review by kfox1111 — COMMENTED** [review]
One more thing I think to make debugging easier and this looks good to me.

File: .github/tests/tornjak/post-install.sh [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+}
+
+cat <<EOF >>"$GITHUB_STEP_SUMMARY"
+### spire
+
+| workload                             | Status |
+| ------------------------------------ | ------ |
+| spire-server                         | "$("${k_rollout_status[@]}" spire-server statefulset spire-server)" |
+| tornjak-frontend                     | "$("${k_wait[@]}" spire-server deployments.apps spire-tornjak-frontend)" |
+| tornjak-frontend                     | "$("${k_rollout_status[@]}" spire-server deployments.apps spire-tornjak-frontend)" |
```
Comment (kfox1111) [inline .github/tests/tornjak/post-install.sh:43]:
```suggestion
| spire-server                         | "$("${k_rollout_status[@]}" spire-server statefulset spire-server)" |
| spire-controller-manager             | "$("${k_rollout_status[@]}" spire-server statefulset spire-controller-manager)" |
| spire-spiffe-oidc-discovery-provider | "$("${k_wait[@]}" spire-server deployments.apps spire-spiffe-oidc-discovery-provider)" |
| spire-spiffe-csi-driver              | "$("${k_rollout_status[@]}" spire-system daemonset spire-spiffe-csi-driver)" |
| spire-agent                          | "$("${k_rollout_status[@]}" spire-system daemonset spire-agent)" |
| tornjak-frontend                     | "$("${k_wait[@]}" spire-server deployments.apps spire-tornjak-frontend)" |
```

File: .github/tests/tornjak/post-install.sh [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+| tornjak-frontend                     | "$("${k_wait[@]}" spire-server deployments.apps spire-tornjak-frontend)" |
+EOF
+
+kubectl -n spire-server get service spire-tornjak-frontend
+
+if [ $1 -ne 0 ]; then
+  get_namespace_details spire-server
+  get_namespace_details spire-system
+fi
+fi
```
Comment (kfox1111) [inline .github/tests/tornjak/post-install.sh:55]:
```suggestion
```

Comment (mrsabath) [reply .github/tests/tornjak/post-install.sh]: Thank you! These merges are painful ...

**Review by kfox1111 — APPROVED** [review]
LGTM. :)

**Pull request merged**